### PR TITLE
milestone 5

### PR DESCRIPTION
**Pull request by evabanegacom**
created associations for friends

### CODE REVIEW

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello,

Thank you for applying some changes. However, there are still some changes to do.

1. You still have to fix the stickler's errors

2. Whatever is the connected user, he still see all the post. Whereas he must only see his own posts and for his friends

3. Clicking on `Add Friend` button is still redirecting me to the user detail page.

4. All the changes requested at the last reviews https://github.com/evabanegacom/ror-social-scaffold/pull/1#pullrequestreview-439229047 are still present.

Kindly make the suggested changes and submit a new code review request once all is OK.
Thanks

Mael FOSSO

**Comment by evabanegacom** [comment]
actually this pr wasn't intended for review at least on my part, i only wanted to be sure i was sending the correct one because i get responses that i sent the wrong pr most times. thanks anyway

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -4,9 +4,24 @@
       <li>
         Name: <%= user.name %>
         <span class="profile-link">
-          <%= link_to 'See Profile',  user_path(user), class: 'profile-link' %>
+          <%= link_to 'See Profile',  user_path(user), class: 'profile-link' %>&nbsp;
+          <% if current_user.friend?(user)%>
```
Comment (uryelah) [inline app/views/users/index.html.erb]: - [ ] All view logic should be inside helpers. Please refactor all conditionals and loops into helper methods.

File: app/views/users/index.html.erb [hunk 2]
```diff
@@ -4,9 +4,24 @@
...
+          <%= link_to 'remove friend', user_remove_path(user), method: :delete, class: 'profile-link' %>
+          <% elsif current_user.pending_friends.include?(user) %>
```
Comment (uryelah) [inline app/views/users/index.html.erb]: - [ ] Avoid using ActiveRecord methods in views, those should be in the models inside methods.

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -1,7 +1,12 @@
 <section class="user-section">
   <h2><%= "Name: #{@user.name}" %></h2>
```
Comment (uryelah) [inline app/views/users/show.html.erb:2]: - [ ] Show all the pending the user friendship requests here if they are the current user, or just the add friend/accept friendship button if it's not the current user.

**Review by uryelah — CHANGES_REQUESTED** [review]
Hi **Precious Udegbue**, I'm Sarah your reviewer for this PR 👩‍🚀 

Good job with most of the asked changes! The touch you added of showing common friends was also very nice. There are still a couple of issues to fix, but read the comments below with care, make sure the changes are there, and you'll be good to go! 💪 

# Required Changes 🏗️ 🐛 💡

- [ ] Improve your PR message, capitalize it, check the grammar describe what you did in this milestone, etc.

- [ ] Don't use Active record queries or methods, like include, order, etc, inside controllers. Those should be inside Active record methods only. (Check PostsController#timeline_posts, LikesController)

- [ ] Show in the posts index not just the user posts, but also the user friends posts.

- [ ] Add tests to your code, for the models and requests.

**Comment by uryelah** [comment]
-----

 **Kindly fix the pointed out issues before requesting the next code review**

------------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Review by Ispirett — APPROVED** [review]
## Status Approved :1st_place_medal: 
Nice work folks with making all the requested changes, you can move on now. :+1:

**Review by whiz25 — APPROVED** [review]
## Approved
Merge and proceed to the next milestone.

### Note: Learn to keep PRs clean. This PR is cluttered with past reviews which are not part of this project. 

### Martin Afani

**Pull request merged**